src/framework/network/inetworkconfiguration.h

```diff
@@ -0,0 +1,38 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2025 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "modularity/imoduleinterface.h"
+
+#include "networktypes.h"
+
+namespace muse::network {
+class INetworkConfiguration : MODULE_EXPORT_INTERFACE
+{
+    INTERFACE_ID(INetworkConfiguration)
+
+public:
+    virtual ~INetworkConfiguration() = default;
+
+    virtual RequestHeaders headers(const std::string& userAgentTemplate = {}) const = 0;
```

Comment:
Maybe it would be better to call it defaultHeaders? 

and as I understand it, it is not the template itself that is transferred here, but only the name
`userAgentTemplate` -> `name` or `userAgentName` or `type` or `userAgentType`